### PR TITLE
Fixes Quash-affected battlers having the wrong order for End Turn effects

### PR DESCRIPTION
**Pull request by PhallenTree**
Similar to how the previous implementation of Quash set the quash affected battler's act order equal to its position and how it didn't swap any battlers' actions, this caused quash affected battlers to have their end of turn effects happen in an order equal to their positions (playerLeft would always be first, playerRight would always be third, etc.) and if users decide to have certain effects not trigger due to having performed a certain action during a turn, these also took into account the wrong battler's action.

See #5400 for more information on these issues and about the previous implementation of Quash.

Also fixes Ally Switch not swapping battlers actions (likely irrelevant since either partner already acted if its action does not match)

## **Discord contact info**
PhallenTree

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -1679,17 +1679,12 @@ u8 DoFieldEndTurnEffects(void)
         switch (gBattleStruct->turnCountersTracker)
         {
         case ENDTURN_ORDER:
-            for (i = 0; i < gBattlersCount; i++)
-            {
-                gBattlerByTurnOrder[i] = i;
-            }
```
Comment (AlexOn1ine) [inline src/battle_util.c:1685]: why was this part removed?

Comment (PhallenTree) [reply src/battle_util.c:1685]: Only changes Turn Order but not actions, also Turn Order should already be correct from the actions taken during the turn.
